### ISSUE TITLE
Crashes after initial login (Pinephone)

### ISSUE THREAD

**Issue by bb83po** [xou816/spot#116]
**Describe the bug**
Crashing after initial login into premium account (Pinephone)

**To Reproduce**
Tried the below steps on both Pinephone and Arch desktop 
1. Open app after installation 
2. Login with premium account 
3. Unable to proceed from here 

**Expected behavior**
To Login successfully and use the app

**Screenshots**
![image](https://user-images.githubusercontent.com/79342490/109473895-32ae2280-7a74-11eb-8bb7-f7da58397fd9.jpeg)

**General information:**
 - Distribution: tried on Mobian(flatpak), arch and Manjaro (both flatpak and aur version)
 - Installation method [e. g. built from source, installed from Flathub...]: flathub and aur 
 - Version [e.g. 0.1.0]:
 - Device used [e. g. desktop, phone...]:

**Stack trace:**
If applicable, run the application from a terminal and paste relevant log output.

```
flatpak run --env=RUST_BACKTRACE=full dev.alextren.Spot
```

**Additional context**
Tried to clear all empty playlists after seeing issue #54. Still no luck.

**Comment by xou816** [comment]
Hi! Thank you for the report :) is it possible to see the first few lines of the stacktrace?

If there's nothing useful here, would you mind installing a snapshot build [from here](https://github.com/xou816/spot/suites/2147100540/artifacts/43949471) and run it as follows:

```
flatpak run --env='RUST_BACKTRACE=full' --branch=master  dev.alextren.Spot
```

**Comment by bb83po** [comment]
Thanks for your quick reply. I ll install the build from there and get back. In the mean time, here is the pic of the first part (sorry in case the quality is not so good). ![image](https://user-images.githubusercontent.com/79342490/109482964-ecaa8c00-7a7e-11eb-9126-65442d61f293.jpeg)

**Comment by Douile** [comment]
https://github.com/xou816/spot/blob/dc877e102c537a0857f9d1bcd4d9e1f533b96b43/src/app/backend/api_models.rs#L271

is now https://github.com/xou816/spot/blob/16a63f0be983d833562b6c5f00362c9db3853ddb/src/app/backend/api_models.rs#L255

I am going to submit a PR to change to a match as SongDescription supports Option for image (this will fix unless it somehow breaks UI).

**Comment by xou816** [comment]
Thank you both!! I'll have a look Douile <3

**Comment by bb83po** [comment]
Thank you very much. Does this mean that I still have a playlist without album artwork ? I went through my full list and unable to find any with blank.

**Comment by Douile** [comment]
From the stack trace I can't really tell what specifically caused the issue. I just prevented the same panic from happening in the future (it was unnecessary panic). 

It was in the conversion from Album to a UI data type so most likely was an album without art work, but could've been something else triggering it.

**Comment by Douile** [comment]
Would you mind compiling from master to see if it fixed, as I might've fixed a symptom not a cause.

**Comment by bb83po** [comment]
Tried using the master branch on my Arch desktop and still the same. After login it crashes. Probably something to do with my account itself ?
[flatpak run log.txt](https://github.com/xou816/spot/files/6061908/flatpak.run.log.txt)


![Spot](https://user-images.githubusercontent.com/79342490/109503198-440a2580-7a9a-11eb-960d-26a16754fcdf.png)

**Comment by xou816** [comment]
you won't be able to update builds from master using the cli directly :/

This is the build with @Douile 's fix : https://github.com/xou816/spot/suites/2148236056/artifacts/43968694

**Comment by Douile** [comment]
^ for some reason the direct link doesn't work for me. You can get the same thing here https://github.com/xou816/spot/actions/runs/610234548 under artifacts

**Comment by bb83po** [comment]
> ^ for some reason the direct link doesn't work for me. You can get the same thing here https://github.com/xou816/spot/actions/runs/610234548 under artifacts

Viola! It's fixed after I built it using the above url. Thanks both of you. Appreciate your quick responses. 

![image](https://user-images.githubusercontent.com/79342490/109522275-6312b280-7aae-11eb-8ada-f0258c48b70e.jpeg)

**Comment by xou816** [comment]
fixed by #117 thanks @Douile :)
